woocommerce/woocommerce-android
# Attempt to fix NPE in gson serialization

Closes: #4868 


### Description


This PR attempts to fix #4868 - more info on [the PR in FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2172). We don't know how to reproduce this issue, so we can't test if this PR fixes the issue or not. However, it'd be good to test that setting `consumerProguardFiles` in FluxC is correctly propagated to the consumer/client app.

### Testing instructions

1. Add "-printconfiguration "build/outputs/mapping/proguard-merged-rules.txt" to the end of `proguard-rules.pro`
2. Run eg. `./gradlew woocommerce:assembleWasabiRelease`
3. When the task finishes open `woocommerce/build/outputs/mapping/proguard-merged-rules.txt` 
4. Notice that the file contains 
```
# Retain generic signatures of TypeToken and its subclasses with R8 version 3.0 and higher.
-keep,allowobfuscation,allowshrinking class com.google.gson.reflect.TypeToken
-keep,allowobfuscation,allowshrinking class * extends com.google.gson.reflect.TypeToken
```

### Images/gif
